### DISCUSSION TITLE
r > zero(r) error using NegativeBinomial

### DISCUSSION THREAD

**Discussion by bhattsamir** [TuringLang/Turing.jl#1789] [Q&A] [answered]
Hi all,

I just wanted to raise a weird issue I have been having when using Turing. Apologies if this is not the right place to discuss this or if I am doing something obviously wrong. I should also note I am new to Turing but really like how you adapted my COVID model a while back. I'd like to do more development using it.

Long story short, using Julia 1.7.2, i have a very simple distribution fitting problem. However, despite r being exponential and positive, and p being beta and between {0,1}, I get a weird error from the NegativeBinomial - 

`NegativeBinomial: the condition r > zero(r) is not satisfied.`

I have tried changing the prior on r (e.g to inverse gamma) but this does not help. The input, data, is simply a matrix of integers. 

code block below

```
M = size(data)[1]
N = size(data)[2]
print(size(data))
@model fit_dist(y) = begin
    r ~  filldist(Exponential(10) ,N)
    p ~  filldist(Beta(1,1),N)
    for j in 1:M
        for i in 1:N
              y[j,i] ~ NegativeBinomial(r[i], p[i])
        end   
    end
end;
fitdist2 = sample(fit_dist(data),  NUTS(), 1000)
```
some more error stack 

`DomainError with Dual{ForwardDiff.Tag{Turing.Essential.var"#f#4"{DynamicPPL.TypedVarInfo{NamedTuple{(:r, :p), Tuple{DynamicPPL.Metadata{Dict{AbstractPPL.VarName{:r, Setfield.IdentityLens}, Int64}, Vector{Product{Continuous, Exponential{Float64}, FillArrays.Fill{Exponential{Float64}, 1, Tuple{Base.OneTo{Int64}}}}}, Vector{AbstractPPL.VarName{:r, Setfield.IdentityLens}}, Vector{Float64}, Vector{Set{DynamicPPL.Selector}}}, DynamicPPL.Metadata{Dict{AbstractPPL.VarName{:p, Setfield.IdentityLens}, Int64}, Vector{Product{Continuous, Beta{Float64}, FillArrays.Fill{Beta{Float64}, 1, Tuple{Base.OneTo{Int64}}}}}, Vector{AbstractPPL.VarName{:p, Setfield.IdentityLens}}, Vector{Float64}, Vector{Set{DynamicPPL.Selector}}}}}, Float64}, DynamicPPL.Model{typeof(fit_dist), (:y,), (), (), Tuple{Matrix{Int64}}, Tuple{}, DynamicPPL.DefaultContext}, DynamicPPL.Sampler{NUTS{Turing.Essential.ForwardDiffAD{40}, (), AdvancedHMC.DiagEuclideanMetric}}, DynamicPPL.DefaultContext}, Float64}}(0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0):`

**Comment by yebai** [comment, marked as answer]
HMC samplers rely on bijective transforms to handle constrained variables (e.g. `r`). If the mass of the constrained variable is close to the boundary, then these transforms might suffer from numerical issues (e.g. violating the support). In the example above, one potential cause is that the prior over `r` has most of its mass close to zero. A simple solution to these problems is using a less informative prior, e.g. `Exponential(0.5)`.

**Reply by bhattsamir** [reply]
makes sense, sadly the change in prior does not help, but I can find another solution now that I know the problem. Thanks

**Reply by mashu** [reply]
@bhattsamir I encounter similar issue with just single variable and I also tried logistic transformation, did you solve your issue? How?

**Reply by bhattsamir** [reply]
Not really, but alot of these problems were due to some mistake in y being wrong (non integer etc). I would check that again. Otherwise the suggestion above from Yebai is all i can suggest
